lib/urldata.h

```diff
@@ -104,7 +104,7 @@ typedef unsigned int curl_prot_t;
 #define PROTO_FAMILY_SSH  (CURLPROTO_SCP|CURLPROTO_SFTP)
 
 #if !defined(CURL_DISABLE_FTP) || defined(USE_SSH) ||   \
-  !defined(CURL_DISABLE_POP3)
+  !defined(CURL_DISABLE_POP3) || !defined(CURL_DISABLE_FILE)
```

Comment:
This enables `CURLOPT_DIRLISTONLY` but acts the same way no matter config. I suppose that is fine, because we can make it do a richer dir listing later. But with this change, I think the documentation for `CURLOPT_DIRLISTONLY` and the corresponding cmdline option (`--list-only`) need to now mention FILE as well